src/main/java/io/github/jancieslak/tba/services/ReservationService.java

```diff
@@ -0,0 +1,151 @@
+                              ScreeningRepository screeningRepository,
+                              ReservationRepository reservationRepository,
+                              ScreeningRoomRepository screeningRoomRepository) {
+        this.reservedSeatsRepository = reservedSeatsRepository;
+        this.screeningRepository = screeningRepository;
+        this.reservationRepository = reservationRepository;
+        this.screeningRoomRepository = screeningRoomRepository;
+    }
+
+    // Check for single place between already reserved seats
+    // If there is such place the seats placement is not valid otherwise it's valid
+    public boolean hasValidSeatsPlacement(Set<SeatReservationModel> seats, Set<ReservedSeat> reservedSeats, Screening screening) {
+        // Get request seats columns grouped by row
+        var partitionedRequestSeats = seats.stream()
+                .collect(Collectors.groupingBy(SeatReservationModel::getRow));
+
+        // Get reserved seats columns filtered by request seats rows
+        var partitionedReservedSeats = reservedSeats.stream()
+                .filter(reservedSeat -> partitionedRequestSeats.get(reservedSeat.getSeat().getRow().getScreeningRoomRow()) != null)
+                .collect(Collectors.groupingBy(reservedSeat -> reservedSeat.getSeat().getRow().getScreeningRoomRow()));
+
+        var screeningRoomOption = screeningRoomRepository.findScreeningRoomByScreeningId(screening.getId());
+
+        if (screeningRoomOption.isEmpty()) {
+            return false;
+        }
+
+        var screeningRoom = screeningRoomOption.get();
+
+        // Assemble a grid of blank seats and reserved/requested seats
+        // If a place is reserved or in requested seats it's true otherwise it's false
+        var screeningRoomRows = screeningRoom.getRows().stream()
+                .filter(row -> partitionedRequestSeats.get(row.getScreeningRoomRow()) != null)
+                .map(row -> {
+                    var requestSeatsSet = partitionedRequestSeats.get(row.getScreeningRoomRow()).stream()
+                            .map(SeatReservationModel::getCol)
+                            .collect(Collectors.toSet());
+                    var reservedSeatsSetOption = partitionedReservedSeats.get(row.getScreeningRoomRow());
+                    Set<Integer> reservedSeatsSet = null;
+
+                    if (reservedSeatsSetOption != null) {
+                        reservedSeatsSet = reservedSeatsSetOption.stream()
+                                .map(reservedSeat -> reservedSeat.getSeat().getSeatCol())
+                                .collect(Collectors.toSet());
+                    }
+
+                    List<Boolean> result = new ArrayList<>();
+
+                    for (int i = 0; i < row.getSeats().size(); i++) {
+                        int col = i + 1;
+
+                        if (requestSeatsSet.contains(col) || (reservedSeatsSet != null && reservedSeatsSet.contains(col))) {
```

Comment:
czy ten if jest potrzebny? czy da się go napisać inaczej?

Reply:
Wydaje mi się że jest potrzebny, a jeśli chodzi o inny sposób napisania tego if-a to jedyną część, którą bym może wyjął poza for to reservedSeatsSet != null który nie musiałby być sprawdzany co iterację